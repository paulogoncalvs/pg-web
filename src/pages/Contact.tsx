import { h, FunctionalComponent } from 'preact';
import { useState } from 'preact/hooks';
import { useTranslate } from '@/modules/i18n';
import { Form } from '@/components/Form';
import { FormInput } from '@/components/Form/Input';
import { Button } from '@/components/Button';

interface ContactForm {
    firstName: string;
    lastName: string;
    email: string;
}

const Contact: FunctionalComponent = () => {
    const [message, setMessage] = useState('');
    const { t } = useTranslate();
    const initialValues: ContactForm = {
        firstName: '',
        lastName: '',
        email: '',
    };

    const submit = ({ firstName, lastName }: ContactForm): void => {
        setMessage(`Hi ${firstName} ${lastName}! This form is still in development. Please come back soon. :)`);
    };

    return (
        <div class="container px-6 pt-8 pb-8 mx-auto sm:pb-16">
            <div class="flex flex-col items-center">
                <h1 class="text-3xl font-black text-center opacity-0 sm:my-2 sm:text-5xl animate-fade-in-up-1">
                    {t('contact_page_title')}
                </h1>
                <h2 class="text-2xl text-center lowercase opacity-0 sm:text-4xl animate-fade-in-up-2 pb-14">
                    {t('contact_page_subtitle')}
                </h2>
            </div>
            <div class="flex flex-col items-center sm:pt-14 pb-14">
                <Form action={`${process.env.SERVER_URL}contact/`} onSubmit={submit} initialValues={initialValues}>
                    <FormInput label="First Name" type="text" name="firstName" id="firstName" />
                    <FormInput label="Last Name" type="text" name="lastName" id="lastName" />
                    <FormInput label="Email Address" type="email" name="email" id="email" />
                    <Button classes="_prim mt-4">{t('contact_page_submit_CTA')}</Button>
                </Form>
                {message && <p class="mt-4">{message}</p>}
            </div>
        </div>
    );
};

export default Contact;
