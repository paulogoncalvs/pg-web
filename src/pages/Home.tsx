import { h, FunctionalComponent } from 'preact';
import { Icon } from '@/components/Icon';
import { Image } from '@/components/Image';
import { useTranslate } from '@/modules/i18n';
import meImg from '@/assets/img/me.jpeg';
import meSmImg from '@/assets/img/me_sm.jpeg';
import mailIcon from '@/assets/icons/mail.svg';

export const Home: FunctionalComponent = () => {
    const { t } = useTranslate();

    return (
        <div class="container mx-auto px-6 py-4 sm:py-8">
            <div class="flex flex-col items-center text-center">
                <h1 class="sm:my-2 text-2xl sm:text-5xl font-black animate-fade-in-up-1 opacity-0">PAULO GONÃ‡ALVES</h1>
                <h2 class="text-xl sm:text-4xl animate-fade-in-up-2 opacity-0 pb-14 lowercase">Front-End Engineer</h2>
                <Image
                    classes="rounded-full"
                    src={meImg}
                    srcset={`${meSmImg}, ${meImg} 2x`}
                    width="230"
                    height="230"
                    alt=""
                    isLazy
                />
                <div class="py-14 text-center space-y-2 text-sm sm:text-base">
                    <p>{t('home_page_description_1')}</p>
                    <p>{t('home_page_description_2')}</p>
                    <p>
                        {t(
                            'home_page_description_3',
                            {
                                link: (
                                    <a
                                        href="https://pt.linkedin.com/in/paulogoncalvs"
                                        target="_blank"
                                        rel="noopener noreferrer"
                                        class="underline"
                                    >
                                        {t('home_page_description_3_link_text')}
                                    </a>
                                ),
                            },
                            false,
                        )}
                    </p>
                </div>
                <div class="pb-14">
                    <h2 class="text-base sm:text-xl font-bold pb-4">{t('home_page_email_button_title')}</h2>
                    <a
                        href="mailto:contact@paulogoncalves.dev"
                        target="_blank"
                        rel="noopener noreferrer"
                        class="transition transform duration-100 hover:translate-x-1 inline-flex px-4 py-2 sm:py-3 items-center justify-center rounded-full shadow-md bg-gradient-to-r from-green-300 via-green-100 to-green-100 hover:from-green-300 hover:via-green-050 hover:to-green-050"
                    >
                        <Icon src={mailIcon} classes="" ariaHidden />
                        <span class="pl-2 sm:text-xl text-gray-900 dark:text-gray-900">
                            {t('home_page_email_button_label')}
                        </span>
                    </a>
                </div>
            </div>
        </div>
    );
};
