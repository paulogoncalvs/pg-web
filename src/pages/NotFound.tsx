import { h, Fragment, FunctionalComponent } from 'preact';
import { useTranslate } from '@/modules/i18n';
import { useLanguage } from '@/modules/language';
import { Link } from '@/components/Link';

const NotFoundPage: FunctionalComponent = () => {
    const { t } = useTranslate();
    const { lang } = useLanguage();

    return (
        <Fragment>
            <div class="flex flex-col items-center">
                <h1 class="text-3xl tracking-tight font-bold text-center opacity-1 sm:my-2 sm:text-5xl animate-fade-in-up-1">
                    {t('not_found_page_title')}
                </h1>
                <h2 class="text-2xl tracking-tight text-center lowercase opacity-1 sm:text-3xl animate-fade-in-up-2 pb-14">
                    {t('not_found_page_subtitle')}
                </h2>
                <div class="opacity-1 animate-fade-in-dw-3">
                    <Link useRouter class="btn _prim" href={`/${lang}/`}>
                        <span>{t('not_found_page_button_label')}</span>
                    </Link>
                </div>
            </div>
        </Fragment>
    );
};

export default NotFoundPage;
