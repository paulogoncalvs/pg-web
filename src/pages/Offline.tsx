import { h, Fragment, FunctionalComponent } from 'preact';
import { useTranslate } from '@/modules/i18n';
import { useLanguage } from '@/modules/language';
import { Link } from '@/components/Link';
import { Markup } from '@/components/Markup';

const OfflinePage: FunctionalComponent = () => {
    const { t } = useTranslate();
    const { lang } = useLanguage();

    return (
        <Fragment>
            <div class="flex flex-col items-center">
                <h1 class="text-3xl tracking-tight font-bold text-center opacity-1 sm:my-2 sm:text-5xl animate-fade-in-up-1">
                    {t('offline_page_title')}
                </h1>
                <h2 class="text-2xl tracking-tight text-center lowercase opacity-1 sm:text-3xl animate-fade-in-up-2 pb-8">
                    {t('offline_page_subtitle')}
                </h2>
                <Markup
                    data={t('offline_page_description')}
                    Element="p"
                    classes="text-l opacity-1 animate-fade-in-dw-4 pb-14 text-center"
                />
                <div class="opacity-1 animate-fade-in-dw-3">
                    <Link useRouter class="btn _prim" href={`/${lang}/`}>
                        <span>{t('offline_page_button_label')}</span>
                    </Link>
                </div>
            </div>
        </Fragment>
    );
};

export default OfflinePage;
