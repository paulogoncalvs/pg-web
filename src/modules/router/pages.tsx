import { h } from 'preact';
import routesConfig from '@/config/routes/index.js';
import Home from '@/pages/Home';
import Contact from '@/pages/Contact';
import NotFound from '@/pages/NotFound';

const pages = {
    Home,
    Contact,
    NotFound,
};

export const getPage = (url: string): JSX.Element => {
    const route = routesConfig[url];
    // @ts-ignore
    const Page = pages[route?.templateParameters?.View] || NotFound;

    return <Page />;
};
