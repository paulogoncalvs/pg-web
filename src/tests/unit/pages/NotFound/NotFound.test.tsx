import { pageTests } from '@/tests/unit/pages';

pageTests({
    name: 'Not Found',
    route: '/404/',
});
