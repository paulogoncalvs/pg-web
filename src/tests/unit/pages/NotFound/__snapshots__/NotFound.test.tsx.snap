// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[PAGE] Not Found Page should render 1`] = `
<main>
  <header class="container p-4 py-10 mx-auto">
    <div class="flex items-center justify-between">
      <div class="flex items-center">
        <select class="py-1 text-sm text-zinc-900 dark:text-zinc-200 bg-white dark:bg-zinc-900 border-zinc-800 dark:border-zinc-200 border-2 font-sans ml-2 mr-2"
                aria-label="Language selection"
        >
          <option value="en"
                  selected
          >
            EN
          </option>
          <option value="pt">
            PT
          </option>
        </select>
      </div>
      <div class="flex justify-end">
        <a href="https://www.twitter.com/paulogoncalvs"
           aria-label="Twitter (opens in a new window)"
           target="_blank"
           rel="noopener noreferrer"
           class="ic-link _hdr"
        >
          <svg aria-hidden="true"
               class="fill-current"
               width="24"
               height="24"
          >
            <use href="#">
            </use>
          </svg>
        </a>
        <a href="https://www.github.com/paulogoncalvs"
           aria-label="GitHub (opens in a new window)"
           target="_blank"
           rel="noopener noreferrer"
           class="ic-link _hdr"
        >
          <svg aria-hidden="true"
               class="fill-current"
               width="24"
               height="24"
          >
            <use href="#">
            </use>
          </svg>
        </a>
        <a href="https://pt.linkedin.com/in/paulogoncalvs"
           aria-label="LinkedIn (opens in a new window)"
           target="_blank"
           rel="noopener noreferrer"
           class="ic-link _hdr"
        >
          <svg aria-hidden="true"
               class="fill-current"
               width="24"
               height="24"
          >
            <use href="#">
            </use>
          </svg>
        </a>
      </div>
    </div>
  </header>
  <div class="container px-6 pt-8 pb-8 mx-auto sm:pb-16">
    <div class="flex flex-col items-center">
      <h1 class="text-3xl font-black text-center opacity-0 sm:my-2 sm:text-5xl animate-fade-in-up-1">
        Oops!
      </h1>
      <h2 class="text-2xl text-center lowercase opacity-0 sm:text-4xl animate-fade-in-up-2 pb-14">
        Page not found
      </h2>
      <a href="/en/"
         class="opacity-0 btn _prim animate-fade-in-dw-3"
      >
        <span>
          Go to homepage
        </span>
      </a>
    </div>
  </div>
</main>
<footer>
  <div class="text-center bg-zinc-100 dark:bg-zinc-800">
    <div class="container flex flex-col items-center px-6 pt-16 mx-auto pb-14">
      <div class="pb-6 text-sm sm:text-base">
        This website was built from scratch with:
      </div>
      <div class="flex flex-wrap justify-center align-middle">
        <a href="https://www.typescriptlang.org/"
           aria-label="TypeScript (opens in a new window)"
           target="_blank"
           rel="noopener noreferrer"
           class="ic-link _ftr"
        >
          <svg aria-hidden="true"
               class="fill-current"
               width="32"
               height="32"
          >
            <use href="#">
            </use>
          </svg>
        </a>
        <a href="https://preactjs.com/"
           aria-label="Preact (opens in a new window)"
           target="_blank"
           rel="noopener noreferrer"
           class="ic-link _ftr"
        >
          <svg aria-hidden="true"
               class="fill-current"
               width="32"
               height="32"
          >
            <use href="#">
            </use>
          </svg>
        </a>
        <a href="https://webpack.js.org/"
           aria-label="Webpack (opens in a new window)"
           target="_blank"
           rel="noopener noreferrer"
           class="ic-link _ftr"
        >
          <svg aria-hidden="true"
               class="fill-current"
               width="32"
               height="32"
          >
            <use href="#">
            </use>
          </svg>
        </a>
        <a href="https://yarnpkg.com/"
           aria-label="Yarn (opens in a new window)"
           target="_blank"
           rel="noopener noreferrer"
           class="ic-link _ftr"
        >
          <svg aria-hidden="true"
               class="fill-current"
               width="32"
               height="32"
          >
            <use href="#">
            </use>
          </svg>
        </a>
        <a href="https://eslint.org/"
           aria-label="ESLint (opens in a new window)"
           target="_blank"
           rel="noopener noreferrer"
           class="ic-link _ftr"
        >
          <svg aria-hidden="true"
               class="fill-current"
               width="32"
               height="32"
          >
            <use href="#">
            </use>
          </svg>
        </a>
        <a href="https://stylelint.io/"
           aria-label="StyleLint (opens in a new window)"
           target="_blank"
           rel="noopener noreferrer"
           class="ic-link _ftr"
        >
          <svg aria-hidden="true"
               class="fill-current"
               width="32"
               height="32"
          >
            <use href="#">
            </use>
          </svg>
        </a>
        <a href="https://developers.google.com/web/tools/workbox"
           aria-label="Workbox (opens in a new window)"
           target="_blank"
           rel="noopener noreferrer"
           class="ic-link _ftr"
        >
          <svg aria-hidden="true"
               class="fill-current"
               width="32"
               height="32"
          >
            <use href="#">
            </use>
          </svg>
        </a>
        <a href="https://analytics.google.com/"
           aria-label="Google Analytics (opens in a new window)"
           target="_blank"
           rel="noopener noreferrer"
           class="ic-link _ftr"
        >
          <svg aria-hidden="true"
               class="fill-current"
               width="28"
               height="28"
          >
            <use href="#">
            </use>
          </svg>
        </a>
        <a href="https://tailwindcss.com/"
           aria-label="Tailwind CSS (opens in a new window)"
           target="_blank"
           rel="noopener noreferrer"
           class="ic-link _ftr"
        >
          <svg aria-hidden="true"
               class="fill-current"
               width="32"
               height="32"
          >
            <use href="#">
            </use>
          </svg>
        </a>
        <a href="https://playwright.dev/"
           aria-label="Playwright (opens in a new window)"
           target="_blank"
           rel="noopener noreferrer"
           class="ic-link _ftr"
        >
          <svg aria-hidden="true"
               class="fill-current"
               width="32"
               height="32"
          >
            <use href="#">
            </use>
          </svg>
        </a>
        <a href="https://jestjs.io/"
           aria-label="Jest (opens in a new window)"
           target="_blank"
           rel="noopener noreferrer"
           class="ic-link _ftr"
        >
          <svg aria-hidden="true"
               class="fill-current"
               width="32"
               height="32"
          >
            <use href="#">
            </use>
          </svg>
        </a>
        <a href="https://prettier.io/"
           aria-label="Prettier (opens in a new window)"
           target="_blank"
           rel="noopener noreferrer"
           class="ic-link _ftr"
        >
          <svg aria-hidden="true"
               class="fill-current"
               width="32"
               height="32"
          >
            <use href="#">
            </use>
          </svg>
        </a>
        <a href="https://postcss.org/"
           aria-label="PostCSS (opens in a new window)"
           target="_blank"
           rel="noopener noreferrer"
           class="ic-link _ftr"
        >
          <svg aria-hidden="true"
               class="fill-current"
               width="32"
               height="32"
          >
            <use href="#">
            </use>
          </svg>
        </a>
        <a href="https://www.deque.com/axe/"
           aria-label="Axe (opens in a new window)"
           target="_blank"
           rel="noopener noreferrer"
           class="ic-link _ftr"
        >
          <svg aria-hidden="true"
               class="fill-current"
               width="32"
               height="32"
          >
            <use href="#">
            </use>
          </svg>
        </a>
        <a href="https://sass-lang.com/"
           aria-label="Sass (opens in a new window)"
           target="_blank"
           rel="noopener noreferrer"
           class="ic-link _ftr"
        >
          <svg aria-hidden="true"
               class="fill-current"
               width="32"
               height="32"
          >
            <use href="#">
            </use>
          </svg>
        </a>
        <a href="https://babeljs.io/"
           aria-label="Babel (opens in a new window)"
           target="_blank"
           rel="noopener noreferrer"
           class="ic-link _ftr"
        >
          <svg aria-hidden="true"
               class="fill-current"
               width="60"
               height="32"
          >
            <use href="#">
            </use>
          </svg>
        </a>
      </div>
    </div>
  </div>
  <div class="container flex flex-col items-center py-8 mx-auto text-xs text-center sm:py-16 text-zinc-500">
    <p>
      optimized for modern browsers
    </p>
    <p class="pt-2">
      paulogoncalves.dev Â© 2021  - 2022 ðŸ™‚
    </p>
  </div>
</footer>
`;
